### DISCUSSION TITLE
Using observe with ViewStore

### DISCUSSION THREAD

**Discussion by tunchamroeun** [pointfreeco/swift-composable-architecture#1413] [Q&A] [answered]
Is the any way or guide to use `observe` directly with `ViewStore` like `WithViewStore(store, observe: ViewState.init){..}`?

#Problem

Compiler error with large view. I want to use `ViewStore` instead and still need to scope state in `WithViewStore(store, observe: ViewState.init){..}`

**Comment by mbrandonw** [comment, marked as answer]
Hi @tunchamroeun, currently you can do this by using `scope` on the store:

```swift
let viewStore = ViewStore(store.scope(state: ViewState.init))
```

But it was an oversight of ours to not add an initializer to `ViewStore` like we did for `WithViewStore`. We will correct that soon.

**Reply by tunchamroeun** [reply]
Thanks for your quick response.

**Comment by tunchamroeun** [comment]
> ```swift
> 
> let viewStore = ViewStore(store.scope(state: ViewState.init))
> 
> ```
From [0.42.0](https://github.com/pointfreeco/swift-composable-architecture/releases/tag/0.42.0)

```swift
let viewStore = ViewStore(store, observe: ViewState.init)
```
